### ISSUE TITLE
quickfix entries don't jump to the correct buffer in multifile project

### ISSUE THREAD

**Issue by teto** [lervag/vimtex#585]
### Explain the issue

Before trying to create a Minimal Working Example, I would like to ask if it's normal that in a multifile project, vimtex doesn't jump to the correct buffer when skipping through the quickfix entries ?
Does it work on vim ? I use neovim and the "subfiles" package.

**Comment by graidl** [comment]
I just wanted to report the same problem, with classical vim, independent of "subfiles" (just using \input) under Linux as well as cygwin. Actually, the messages in the quickfix window show the the main file instead of the included file where the problem arises. I encounter this for example with "underful \hbox" or "undefined reference" warnings.

**Comment by lervag** [comment]
In most cases, this should work. However, I know that it won't work in all cases. This sucks.

The reason is that it is extremely difficult to parse the log file, largely because of the log file format that often breaks (hard wraps) file paths in the middle. I've tried to handle this as best as possible, but I think in order to handle it more generally, we would need to preprocess the log file with something that provides a cleaner and more easily parsed error/warning output.

However, I would be happy to consider the specific cases where this does not work for you. In the test project under `VIMTEX/test/examples/book-multifile`, one should get errors when compiling at multiple locations. When I open the `main.tex` and start compiling with `\ll`, I get the correct file paths for the errors in the quickfix window.

**Comment by graidl** [comment]
Thanks for your quick reply and explanation, which I fully understand. Latex's output is indeed a huge pain to further process.
The problems I encountered can be easily seen also in your `book-multifile` example when inserting somewhere in e.g. chapters.tex a `\ref{blabla}` without defining this reference somewhere else, or alternatively, a long line raising any `Overfull or Overfull \hbox` warning. 
You then get these warnings:
```
main.tex|6 warning| LaTeX Warning: Reference `blabla' on page 3 undefined on input line 6.
main.tex|5 warning| Overfull \hbox (564.91011pt too wide) in paragraph at lines 5--8
```
The line numbers are fine but obviously the indicated `main.tex` isn't.
However, I can confirm that with other errors, e.g. some undefined command, the correct file is detected. Maybe these is somehow specific to warnings.

As these kinds of warnings are quite common, it would be great to find a solution in particular for them. However, if this is not possible without realizing a full-fledged parser, I can also live well with the current situation. Thanks for your superb work!

**Comment by teto** [comment]
> Thanks for your superb work!

I concur. vimtex is my only light in the vimtex hell (and texstudio as well); Yeah (la)Tex is awful, but it's still the best we got :(
How does vimtex process the log file ? plain vim errorformat ? could preprocessing the log with any of the solution listed here:
http://tex.stackexchange.com/questions/10558/use-latexmk-to-filter-the-log-file
could help ? This could be done via `g:vimtex_preprocess_log_command = ""` or just by replacing latexmk via rubber in the compile command ?
https://github.com/stefanhepp/pplatex and rubber-info look like popular solutions.

For now I edit in vim and compile in texstudio, it seems to have some better log parsing.

**Comment by languitar** [comment]
I am also facing the issue that included files are not recognized at all in the error messages. Here is my compile log: https://gist.github.com/languitar/eef7c8b09104c11ef4663cb0ae5210ae

**Comment by lervag** [comment]
This issue is now covered by #651.

**Comment by lervag** [comment]
@teto Hi. @languitar implemented a quickfix layer for pplatex in #816. It is merged and is now available with `let g:vimtex_quickfix_method = 'pplatex'` (assuming you have pplatex installed, of course).
